samtranslator/swagger/swagger.py

```diff
@@ -695,6 +700,10 @@ def add_auth_to_method(self, path: str, method_name: str, auth: Dict[str, Any],
             self._set_method_authorizer(path, method_name, method_authorizer, authorizers, method_scopes)  # type: ignore[no-untyped-call]
 
         method_apikey_required = auth and auth.get("ApiKeyRequired")
+
+        if auth.get("AddApiKeyRequiredToCorsPreflight") and method_name == "options":
```

Comment:
I don't see any type validation from what I can see, so I'm assuming even `"false"` will evaluate as `true` here?

Reply:
I think this actually doesn't have to be here at all I will remove it and test.

Reply:
Okay, so this is necessary if they define AddApiKeyRequiredToCorsPreflight: false at the individual Api level. I did miss the logic flip so I need to change it to if not auth.get("AddApiKeyRequiredToCorsPreflight") and method_name == "options":

Reply:
Okay, can you add (transform?) tests as well that check this logic?